lib/HttpClient.ts

```diff
@@ -365,17 +367,19 @@ export class HttpClient implements ifm.IHttpClient {
         const usingSsl: boolean = info.parsedUrl.protocol === 'https:';
         info.httpModule = usingSsl ? https : http;
         const defaultPort: number = usingSsl ? 443 : 80;
+        
         info.options = <http.RequestOptions>{};
         info.options.host = info.parsedUrl.hostname;
         info.options.port = info.parsedUrl.port ? parseInt(info.parsedUrl.port) : defaultPort;
         info.options.path = (info.parsedUrl.pathname || '') + (info.parsedUrl.search || '');
-        info.options.method = method;
+        info. options.method = method;
+
         info.options.headers = this._mergeHeaders(headers);
         if (this.userAgent != null) {
             info.options.headers["user-agent"] = this.userAgent;
         }
         
-        info.options.agent = this._getAgent(requestUrl);
```

Comment:
Is there a reason to change this?

Reply:
all the using methods of requestUrl immediately called url.parse() on it, so passing the already parsed Url object seemed like better than re-parsing the string each time.